pkg/controllers/nodetemplate/controller.go

```diff
@@ -0,0 +1,90 @@
+/*
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package nodetemplate
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	controllerruntime "sigs.k8s.io/controller-runtime"
+	k8sClient "sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller"
+	"sigs.k8s.io/controller-runtime/pkg/manager"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+
+	"github.com/aws/aws-sdk-go/service/ec2/ec2iface"
+
+	corecontroller "github.com/aws/karpenter-core/pkg/operator/controller"
+	"github.com/aws/karpenter/pkg/apis/v1alpha1"
+	"github.com/aws/karpenter/pkg/provider"
+)
+
+var _ corecontroller.TypedController[*v1alpha1.AWSNodeTemplate] = (*Controller)(nil)
+
+type Controller struct {
+	kubeClient     k8sClient.Client
+	subnet         *provider.SubnetProvider
+	securityGroups *provider.SecurityGroupProvider
+}
+
+func NewController(client k8sClient.Client, ec2api ec2iface.EC2API, subnetProvider *provider.SubnetProvider, securityGroups *provider.SecurityGroupProvider) corecontroller.Controller {
+
+	return corecontroller.Typed[*v1alpha1.AWSNodeTemplate](client, &Controller{
+		kubeClient:     client,
+		subnet:         subnetProvider,
+		securityGroups: securityGroups,
+	})
+}
+
+func (c *Controller) Reconcile(ctx context.Context, ant *v1alpha1.AWSNodeTemplate) (reconcile.Result, error) {
+	fmt.Println("Controller Here")
+	subnetList, err := c.subnet.Get(ctx, ant)
+	subnetLog := provider.PrettySubnets(subnetList)
+	if err != nil {
+		// Back off and retry reconciliation
+		return reconcile.Result{RequeueAfter: 1 * time.Minute}, err
```

Comment:
This RequeueAfter may be ignored if there is an error returned. I believe we also do an exponential back off per item so that all you need to do is Requeue: true or return err if you want it to just try again. Maybe there was a reason you wanted one minute hard coded?
```suggestion
		return reconcile.Result{}, err
```